projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -41,60 +52,93 @@ class TuiRippleStyles {}
+    };
 
-    @Input()
-    public tuiRipple?: string | '';
+    @Input({
+        alias: 'tuiRipple',
+        transform: selectorScopeFallback,
+    })
+    public selector = DEFAULT_SELECTOR;
 
     protected readonly nothing = tuiWithStyles(TuiRippleStyles);
 
-    constructor() {
-        const touchEnd$ = tuiTypedFromEvent(this.el, 'touchend');
-        const touchMove$ = tuiTypedFromEvent(this.el, 'touchmove');
+    @HostListener('pointerdown.silent', [
+        '$event.clientX',
+        '$event.clientY',
+        '$event.target',
+    ])
+    protected start(x: number, y: number, target: HTMLElement): void {
+        const element = target.closest(this.selector);
 
-        this.end$.subscribe(element => this.renderer.removeChild(this.el, element));
+        if (!tuiIsHTMLElement(element)) {
+            return;
+        }
 
-        this.start$
-            .pipe(
-                mergeMap(ripple => {
-                    const animationEndOn$ = tuiTypedFromEvent(ripple, 'animationend');
+        const ripple = this.createRipple(x, y, element.getBoundingClientRect());
+        const touchEnd$ = merge(
+            tuiTypedFromEvent(element, 'pointerup'),
+            tuiTypedFromEvent(element, 'pointercancel'),
+        );
+        const touchMove$ = tuiTypedFromEvent(element, 'pointermove');
 
-                    return race(
-                        timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER)),
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
-                        }),
-                        switchMap(isTap =>
-                            isTap
-                                ? animationEndOn$
-                                : race(
-                                      touchEnd$.pipe(switchMap(() => animationEndOn$)),
-                                      animationEndOn$.pipe(switchMap(() => touchEnd$)),
-                                  ),
-                        ),
-                        map(() => ripple),
-                    );
-                }),
-                takeUntilDestroyed(),
+        element.appendChild(ripple);
+
+        const animationEndOn$ = tuiTypedFromEvent(
+            ripple.animate(TO_KEYFRAMES, this.animationOptions),
+            'finish',
+        );
+
+        race(
+            touchEnd$.pipe(switchMap(() => animationEndOn$)),
+            animationEndOn$.pipe(switchMap(() => touchEnd$)),
+        )
+            .pipe(
+                first(),
```

Comment:
Let's switch to `take(1)` otherwise it will throw an error if component is destroyed while the finger holds it.

Reply:
No, it shouldn't